### PR TITLE
refactor(web): use queries to track the installer status

### PR DESCRIPTION
**Pull request by imobachgs**
Follow-up of #1439. The goal of this PR is to change installer status tracking to Tanstack Query. Additionally, it adjust the HTTP API a bit.

## Notes

The `WithStatus` mixin should not be relevant anymore. However, the storage UI still uses it to detect when the proposal changes. We should use a dedicated event instead.

### CODE REVIEW

File: web/src/App.jsx [hunk 1]
```diff
@@ -28,9 +28,9 @@ import { useInstallerL10n } from "./context/installerL10n";
 import { useInstallerClientStatus } from "~/context/installer";
 import { useProduct, useProductChanges } from "./queries/software";
 import { CONFIG, INSTALL, STARTUP } from "~/client/phase";
-import { BUSY } from "~/client/status";
```
Comment (dgdavid) [inline web/src/App.jsx:31]: As discussed offline, please check if _client/status_ still needed.

File: web/src/App.jsx [hunk 2]
```diff
@@ -64,15 +66,15 @@ function App() {
         </SimpleLayout>
       );
 
-    if ((phase === STARTUP && status === BUSY) || phase === undefined || status === undefined) {
```
Comment (dgdavid) [inline web/src/App.jsx:67]: I understand that `phase === undefined` is either, not possible anymore or nor relevant here. Could you please confirm it?

File: web/src/components/core/Installation.jsx [hunk 3]
```diff
@@ -21,10 +21,9 @@
 
 import React from "react";
 import { InstallationProgress, InstallationFinished } from "~/components/core";
-import { IDLE } from "~/client/status";
 
-function Installation({ status }) {
-  return status === IDLE ? <InstallationFinished /> : <InstallationProgress />;
+function Installation({ isBusy }) {
+  return isBusy ? <InstallationFinished /> : <InstallationProgress />;
```
Comment (dgdavid) [inline web/src/components/core/Installation.jsx:26]: NP: out of the scope of the PR, but you might try to move small components and their tests to TypeScript.

File: web/src/components/overview/SoftwareSection.test.tsx [hunk 4]
```diff
@@ -32,7 +32,7 @@ let mockTestingProposal: SoftwareProposal;
 jest.mock("~/queries/software", () => ({
   usePatterns: () => mockTestingPatterns,
   useProposal: () => mockTestingProposal,
-  useProposalChanges: jest.fn(),
+  useProposalChanges: () => jest.fn(),
```
Comment (dgdavid) [inline web/src/components/overview/SoftwareSection.test.tsx:35]: good catch! :+1:

File: web/src/components/product/ProductSelectionProgress.jsx [hunk 5]
```diff
@@ -35,15 +36,9 @@ import { PATHS } from "~/router";
...
-  const [status, setStatus] = useState();
+  const { isBusy } = useInstallerStatus({ suspense: true });
 
-  useEffect(() => {
-    manager.getStatus().then(setStatus);
-    return manager.onStatusChange(setStatus);
-  }, [manager, setStatus]);
-
-  if (status === IDLE) return <Navigate to={PATHS.root} replace />;
+  if (!isBusy) return <Navigate to={PATHS.root} replace />;
```
Comment (dgdavid) [inline web/src/components/product/ProductSelectionProgress.jsx:41]: That was a good movement. I like to see how components are getting cleaner little by little.

File: web/src/types/status.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+type InstallerStatus = {
+  /** Whether the installer is busy */
+  isBusy: boolean;
+  /** Installation phase */
+  phase: InstallationPhase;
+  /** Whether the installation can be performed or not */
+  canInstall: boolean;
+  /** Whether the installer is running on Iguana */
+  useIguana: boolean;
+};
```
Comment (dgdavid) [inline web/src/types/status.ts:43]: :ok_hand:

**Review by dgdavid — APPROVED** [review]
It looks really good, thanks!

File: web/src/App.jsx (hunk 2 above)
Comment (imobachgs) [reply web/src/App.jsx:67]: Yes, that's it. We are using `suspense: true` to get the installer status.

File: web/src/App.jsx (hunk 1 above)
Comment (imobachgs) [reply web/src/App.jsx:31]: It is still needed by the storage part. But `client/phase` is not, so I have removed it.

File: web/src/components/core/Installation.jsx (hunk 3 above)
Comment (imobachgs) [reply web/src/components/core/Installation.jsx:26]: I would prefer to do that on a separate PR.

**Pull request merged**